src/software/new_geom/segment.cpp

```diff
@@ -0,0 +1,51 @@
+#include "software/new_geom/segment.h"
+
+void Segment::setStart(Point o)
+{
+    start = o;
+}
+
+const Point& Segment::getStart() const
+{
+    return start;
+}
+
+void Segment::setEnd(Point o)
+{
+    end = o;
+}
+
+const Point& Segment::getEnd() const
+{
+    return end;
+}
+
+double Segment::length() const
+{
+    return (end - start).length();
+};
+
+double Segment::lengthSquared() const
+{
+    return (end - start).lengthSquared();
+}
+
+Segment Segment::reverse() const
+{
+    return Segment(end, start);
+};
+
+Vector Segment::toVector() const
+{
+    return end - start;
+};
+
+double Segment::slope() const
+{
+    return (end.y() - start.y()) / (end.x() - start.x());
+};
```

Comment:
Where is this `slope` function used? My concern is that it's not clear if it's dx/dy or dy/dx and isn't super robust to degenerate cases, like `dx` being 0. Ideally we just stop using it I think?

Reply:
I don't think slope is being used anywhere, so I'm removing it.